### PR TITLE
(PDB-840) Retire deprecated commands

### PR DESCRIPTION
**Pull request by ajroetker**
This commit retires all of the previously deprecated commands and
removes all the associated support code for these commands, such as
validation of reports/catalogs for the deprecated versions. This commit
also updatees al the unit/integration tests to use the non-retired
versions of these commands and removes test code we no longer use.

### CODE REVIEW

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by wkalt** [comment]
@pljenkinsro retest this please

**Comment by kbarber** [comment]
@pljenkinsro retest this please.

@ajroetker rerunning the tests here again, looks like the acceptance failures where just a ghprb bug, but the travis failures look like something else.

**Comment by kbarber** [comment]
@pljenkinsro retest this please

**Comment by kbarber** [comment]
@ajroetker still waiting on PR testing for acceptance, but looks like the ruby rspec tests are at least failing on this one for now: https://travis-ci.org/puppetlabs/puppetdb/jobs/45724423 I'm guessing its the changes to the storeconfigs command that are causing this.

**Comment by kbarber** [comment]
@pljenkinsro retest this please

Damn ghprb bug again, weird.

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by wkalt** [comment]
@pljenkinsro retest this please

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

File: src/puppetlabs/puppetdb/cli/services.clj [hunk 1]
```diff
@@ -337,11 +336,18 @@
...
    [:ShutdownService shutdown-on-error]]
 
   (start [this context]
-         (start-puppetdb context (get-config) this add-ring-handler shutdown-on-error))
+         (let [config (get-config)
+               configure-global-url-prefix
+               (fn [global-config]
+                 (let [webrouter-url-prefix (get-route this)]
+                   (when (seq (:url-prefix global-config))
+                     (log/warn "The configuration item `url-prefix` in the [global] section is deprecated. It will be removed in the future.")
```
Comment (senior) [inline src/puppetlabs/puppetdb/cli/services.clj]: How does this relate to https://github.com/puppetlabs/puppetdb/pull/1198 ?

Comment (ajroetker) [inline src/puppetlabs/puppetdb/cli/services.clj]: Looks like there was an accidental `git add`, nice catch.

**Comment by wkalt** [comment]
This looks great to me. There are some references to old wire-format versions as "deprecated" in index.markdown -- maybe these should be changed to "retired" or removed altogether?

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Comment by wkalt** [comment]
In documentation/api/wire_format/facts_format_v3.markdown there's a reference to "differences from v2", which seems a bit weird since v2 isn't documented anymore. Of course, it'll be weird that it starts at v3 now too -- seems to me like the consistent thing to do is remove references at this point though.

Also, in commands.markdown, there are a bunch of links at the top & used throughout that will be invalid once this patch deletes their targets.

**Comment by wkalt** [comment]
There are also references to previous versions in 
./resources/ext/docs/api/wire_format/catalog_format_v5.markdown
and 
./resources/ext/docs/api/wire_format/report_format_v4.markdown

**Comment by ajroetker** [comment]
@wkalt Alright i removed all those too!

**Comment by wkalt** [comment]
+1

**Comment by ajroetker** [comment]
@pljenkinsro retest this please

**Pull request merged**